integration/hive/src/main/java/org/apache/carbondata/hive/util/DataTypeUtil.java

```diff
@@ -21,25 +21,31 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import org.apache.carbondata.core.constants.CarbonCommonConstants;
 import org.apache.carbondata.core.metadata.datatype.DataType;
 import org.apache.carbondata.core.metadata.datatype.DataTypes;
 import org.apache.carbondata.core.metadata.datatype.StructField;
 
+import org.apache.commons.lang.ArrayUtils;
+
 public class DataTypeUtil {
 
   public static DataType convertHiveTypeToCarbon(String type) throws SQLException {
     if ("string".equalsIgnoreCase(type) || type.startsWith("char")) {
       return DataTypes.STRING;
-    } else if ("varchar".equalsIgnoreCase(type)) {
+    } else if (!type.startsWith("map<") && !type.startsWith("array<") && !type.startsWith("struct<")
```

Comment:
why map, struct, array check required?

Reply:
made required changes, they are not required